test/unit/int_algolia/scripts/algolia/helper/productFilter.test.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var ProductMock = require('../../../../../mocks/dw/catalog/Variant');
+
+describe('productFilter.isIncludeOutOfStock', () => {
+    var productFilter = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/filters/productFilter');
+    const AlgoliaLocalizedProductModel = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct');
+
+    beforeEach(() => {
+        jest.resetModules();
+    });
+
+    test('skips out-of-stock if IndexOutOfStock = false', () => {
+        jest.doMock('*/cartridge/scripts/algolia/lib/algoliaData', () => ({
+            getSetOfArray: () => [],
+        }), { virtual: true });
+
+
+        const product = new ProductMock();
+        const inStockVariant =  new AlgoliaLocalizedProductModel({ 
+            product: product,
+            locale: 'default',
+            attributeList: ['custom.displaySize']
+        });
+
+        inStockVariant.getAvailabilityModel = () => {
+            return {
+                getInventoryRecord: () => {
+                    return { getATS: () => { return { getValue: () => { return 5; } } } };
+                },
+            };
+        }
+        const inStock = productFilter.isInStock(inStockVariant, 1);
```

Comment:
Oh, I hadn't really looked at the content of the test. It's not good:
Your `isInStock()` method expects a SFCC product in parameter.
Here, you are instantiating a SFCC product, but then:
- you use the SFCC product to instantiate an `AlgoliaLocalizedProduct`
- you declare the `getAvailabilityModel` method on our `AlgoliaLocalizedProduct` object
- you use this object to test the `isInStock()` method

It works, because JavaScript is flexible, but it's not correct.
What should be done is either:
- Override the method directly on the `product` instance
- Or improve the `VariantMock` to be able to pass the ATS in the constructor